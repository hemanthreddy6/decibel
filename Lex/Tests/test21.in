import "old_production.db";

function combine <- (audio1: audio, sound2: audio): audio => (sound1 + sound2);

function valid_album <- (float: average_length, count: int): bool {
    if average_length * count > 20 {
        return true;
    } otherwise {
        return false;
    }
}

function main <- (): int {
    my_music: audio <- load "song1.wav"; // Taking a song as input
    backing_track: audio <- load "bt.wav";

    duration: float <- 3.5;
    count: int <- 4;

    /* Scaling the volume of the
        backing track down */

    backing_track <- backing_track*0.5;

    valid: bool <- valid_album(duration, count);

    new_track: audio <- combine(my_music, backing_track);   // Another comment
    play new_track;
    save new_track -> "song2.wav";

    return 0;
}