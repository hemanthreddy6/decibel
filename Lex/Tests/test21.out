IMPORT
STRING_LITERAL
;
FUNCTION
IDENTIFIER
<-
(
IDENTIFIER
:
AUDIO
,
IDENTIFIER
:
AUDIO
)
:
AUDIO
=>
(
IDENTIFIER
+
IDENTIFIER
)
;
FUNCTION
IDENTIFIER
<-
(
FLOAT
:
IDENTIFIER
,
IDENTIFIER
:
INT
)
:
BOOL
{
IF
IDENTIFIER
*
IDENTIFIER
>
INT_LITERAL
{
RETURN
TRUE
;
}
OTHERWISE
{
RETURN
FALSE
;
}
}
FUNCTION
IDENTIFIER
<-
(
)
:
INT
{
IDENTIFIER
:
AUDIO
<-
LOAD
STRING_LITERAL
;
IDENTIFIER
:
AUDIO
<-
LOAD
STRING_LITERAL
;
IDENTIFIER
:
FLOAT
<-
FLOAT_LITERAL
;
IDENTIFIER
:
INT
<-
INT_LITERAL
;
IDENTIFIER
<-
IDENTIFIER
*
FLOAT_LITERAL
;
IDENTIFIER
:
BOOL
<-
IDENTIFIER
(
IDENTIFIER
,
IDENTIFIER
)
;
IDENTIFIER
:
AUDIO
<-
IDENTIFIER
(
IDENTIFIER
,
IDENTIFIER
)
;
PLAY
IDENTIFIER
;
SAVE
IDENTIFIER
->
STRING_LITERAL
;
RETURN
INT_LITERAL
;
}
